statement.go

```diff
@@ -768,6 +776,31 @@ func (s *DmlStatement) Execute(session *Session) (*Result, error) {
 	return result, nil
 }
 
+type PartitionedDmlStatement struct {
+	Dml string
+}
+
+func (s *PartitionedDmlStatement) Execute(session *Session) (*Result, error) {
+	if session.InRwTxn() {
+		// PartitionedUpdate creates a new transaction and it could cause dead lock with the current running transaction.
+		return nil, errors.New(`Partitioned DML statement can not be run in a read-write transaction`)
+	}
+	if session.InRoTxn() {
+		// Just for user-friendly.
+		return nil, errors.New(`Partitioned DML statement can not be run in a read-only transaction`)
+	}
+
+	stmt := spanner.NewStatement(s.Dml)
+	count, err := session.client.PartitionedUpdate(session.ctx, stmt)
+	if err != nil {
+		return nil, err
+	}
+	return &Result{
+		IsMutation:   true,
+		AffectedRows: int(count),
```

Comment:
> ```
> spanner> partitioned update Songs set SongName = "Foo Bar" WHERE SongGenre = 'ROCK';
> Query OK, 3 rows affected (3.75 sec)
> ```

Currently, affected rows are always shown as an exact value but it is not exactly correct.
How about showing this as an estimated value?

It can say as "lower bound" or "at least" rather than simply "estimated".

https://cloud.google.com/spanner/docs/reference/rpc/google.spanner.v1?hl=en#resultsetstats

>Partitioned DML does not offer exactly-once semantics, so it returns a lower bound of the rows modified.

https://github.com/googleapis/google-cloud-go/blob/spanner/v1.10.0/spanner/read.go#L167-L168

In gcloud:
```
$ gcloud spanner databases execute-sql ${DATABASE_ID} --sql='DELETE FROM tbl WHERE TRUE' --enable-partitioned-dml
Statement modified a lower bound of 0 rows
```

Reply:
That makes sense. I added "at least" for the result of partitioned DML: 657d5f8

Example:
```
spanner> partitioned delete from Songs WHERE SongGenre = 'ROCK';                                     
Query OK, at least 3 rows affected (3.59 sec)
                                                                                                                                                                                                          
spanner> insert into Singers (SingerId, FirstName, LastName) values (6, "a", "b");                                                                                                                        
Query OK, 1 rows affected (0.85 sec)
```